Dropdown - don't stop event propagation

sijusamson opened the pull request:
Fixes #30267

Preview: https://deploy-preview-32943--twbs-bootstrap.netlify.app/

sijusamson committed:
Event bubbling issue

1) Removed Stop Propagation
2) Added conditions for 'Dropdown clear menu' function

XhmikosR commented:
We also have #32625, /CC @rohit2sharma95

rohit2sharma95 commented:
It is about another issue that is when the dropdown toggle button is clicked, the (click) event is not bubbling up because of the `stopPropagation`.

mohe2015 commented:
Does this also fix https://github.com/twbs/bootstrap/issues/30882 ?

XhmikosR commented:
@sijusamson can you add a test please?

XhmikosR commented:
@sijusamson could this possibly fix #33044? If so, can you add another test case covering that too please?

sijusamson commented:
> @sijusamson could this possibly fix #33044? If so, can you add another test case covering that too please?

Not sure. I will have a look.
And I am so sorry for reverting too late :-(

sijusamson committed:
Added a test case for event bubbling

sijusamson committed:
Merge issue resolved

sijusamson committed:
Removed unnecessary conditions

XhmikosR commented on js/tests/unit/dropdown.spec.js:
```diff
@@ -1719,9 +1719,7 @@ describe('Dropdown', () => {
-      '    <a id="item1" class="dropdown-item" href="#">Another link</a>',
-      '  </div>',
-      '</div>'
+      '    <a id="item1" class="dropdown-item" href="#">Another link</a>'
```
This change seems unrelated?

XhmikosR commented on js/tests/unit/dropdown.spec.js:
```diff
@@ -1756,7 +1754,34 @@ describe('Dropdown', () => {
         handleArrowUp()
       }
     })
-
```
This one also

XhmikosR commented on js/tests/unit/dropdown.spec.js:
```diff
@@ -1756,7 +1754,34 @@ describe('Dropdown', () => {
+    triggerDropdown.click()
+  })
 })
+
```
And this one

sijusamson committed:
Removing unnecessary changes

XhmikosR committed:
Update dropdown.spec.js

XhmikosR commented:
Hmm, this seems to break at least our docs version dropdown

![image](https://user-images.githubusercontent.com/349621/107810536-82eb6c00-6d75-11eb-8bfe-5d000c7c2618.png)

sijusamson commented:
> Hmm, this seems to break at least our docs version dropdown
> 
> ![image](https://user-images.githubusercontent.com/349621/107810536-82eb6c00-6d75-11eb-8bfe-5d000c7c2618.png)

Great catch. :100:

sijusamson committed:
Dropdown click

1) Add a condition for child elements to work as expected in dropdown
2) Added a test case for child elements

XhmikosR commented:
@sijusamson can you please add an extra test so that the previous regression is caught? Because I only caught the issue manually.

sijusamson reviewed (commented):
@XhmikosR I have added a test case for the same. The issue was caused because a child element was present within the dropdown. 
I have handled the case as well as added a test case too.

rohit2sharma95 committed:
Also, expect the hidden event not to be emitted

- To make sure that the side effects are not made that wes expected
  when the `clearMenus` method is called more than once

rohit2sharma95 commented:
It does not fix #33044 though

rohit2sharma95 committed:
Merge branch 'main' into dropdown__bubbling

XhmikosR commented:
@sijusamson can you have a look in case you can also fix #33044 too here?

sijusamson commented:
> @sijusamson can you have a look in case you can also fix #33044 too here?

Sure !

XhmikosR commented:
Actually wait, it seems #32942 might be a cleaner solution :)

sijusamson commented:
> Actually wait, it seems #32942 might be a cleaner solution :)

Okay.

sijusamson commented:
Closing this PR as #32942 appears to be a better solution.

arcanisgk commented:
by default there should be no blocking of events in the use of dropdown box; but if you want to add it programmatically it should be possible to insert a block of events.

What would happen if you want to track where the user navigates through the dropdown boxes and the events are blocked ...
